### PR TITLE
enter key behaviour in form improved

### PR DESCRIPTION
**Pull request by magg01**
I noticed an issue with using the 'Enter' key on form input when reviewing our UI. (see linked issue)

This is my attempt to provide more expected behaviour to a user and avoid unexplained data loss. Let me know what you think.

### CODE REVIEW

File: src/Components/Home/NewQuestionModal.js [hunk 1]
```diff
@@ -2,8 +2,26 @@ import React from "react";
...
+
+  const handleKeyboardEvent = (e) => {
+    if(!!props.question && e.keyCode === 13){
+      e.preventDefault();
+      if(props.answer.length !== 0){
+        confirmAnswer();
+      }
+    }
+  }
+
```
Comment (dmytrokulak) [inline src/Components/Home/NewQuestionModal.js]: It is a good catch and seems like a reasonable fix. I don't know whether there is another way to approach this.

As for the code cleanness:  
1) we now have `!!props.question && props.answer.length === 0` in two places - would be nice to have a const for this e.g. hasEmptyAnswer 
2) 13 is a magic string, though a well-know one, but nevertheless. but can also warrant a const e.g. EnterKeyCode

There two are not critical, up to you.

**Review by dmytrokulak — APPROVED** [review]

File: src/Components/Home/NewQuestionModal.js (hunk 1 above)
Comment (magg01) [reply src/Components/Home/NewQuestionModal.js]: Good points. I'll implement both of these.

Comment (magg01) [reply src/Components/Home/NewQuestionModal.js]: actually I don't see the duplication of `!!props.question && props.answer.length === 0` so have left that for now

Comment (dmytrokulak) [reply src/Components/Home/NewQuestionModal.js]: an old one is at line 68, a new one is split between lines 15 and 17 as visible in the code snippet above.  it is more visible if !!props.question is moved into the second if. which doesn't ruin any logic - we can always prevent default for Enter key press, and process confirmation if question is defined and answer is not empty.

**Pull request merged**